app/CodeFormatterApp.php

```diff
@@ -104,6 +110,7 @@ protected function processPostContent()
     protected function splitPostAtCodeBlockEnding()
     {
         // split post at code block ending
+        \assert(\is_string($this->post_content)); // FIXME
         $this->post_content = preg_split('/\[\/CODE\]/', $this->post_content);
```

Comment:
Based on the revised method `readPostContent`, we know for sure that `$this->post_content` is a string if nothing goes terribly wrong internally.
So, is there even any point in checking that?
Sorry, just curious why you do that or rather why phan would still complain here after your revision above.

Reply:
yes, but still, post_content is used as string and array in the current code, thtas why the assertion is there :)
phan does not execute the code, it just analyzes